### PR TITLE
Refactor/file id format, add data partner, searchRequestJob unit tests

### PR DESCRIPTION
**Pull request by peggy-gov-bc-ca**
# Description

This PR includes the following proposed change(s):

FAMS3-2156(Refactor/file id format, add data partner, searchRequestJob unit tests)

## Type of change

- [X] Bug fix (non-breaking change which fixes an issue)
- [] New feature (non-breaking change which adds functionality)
- [X] Refactoring / Documentation

if your change is a breaking change, please add `breaking change` label to this PR

## How Has This Been Tested?

Tested locally

## Checklist:

- [ ] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [X] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: app/DynamicsAdapter/DynamicsAdapter.Web.Test/MockLoggerExtensions.cs [hunk 1]
```diff
@@ -0,0 +1,25 @@
+﻿using Microsoft.Extensions.Logging;
+using Moq;
+using System;
+
+namespace DynamicsAdapter.Web.Test
+{
+    public static class MockLoggerExtensions
```
Comment (rantifamilusi) [inline app/DynamicsAdapter/DynamicsAdapter.Web.Test/MockLoggerExtensions.cs:7]: can we change to MockLoggerExtensionsTest ?

What is this testing, the enricher extensions?

File: app/DynamicsAdapter/DynamicsAdapter.Web.Test/MockLoggerExtensions.cs [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+        {
+            loggerMock.VerifyLog(level, message, Times.Once(), failMessage);
+        }
+        public static void VerifyLog<T>(this Mock<ILogger<T>> loggerMock, LogLevel level, string message, Times times, string failMessage = null)
+        {
+            loggerMock.Verify(l => l.Log(level, It.IsAny<EventId>(),
+                       It.Is<It.IsAnyType>((o, _) => o.ToString() == message), It.IsAny<Exception>(),
+                       It.Is<Func<It.IsAnyType, Exception, string>>((v, t) => true)),
+                   times, failMessage);
+            //  .NET Core 2.1
```
[the remark this answers is not in the document]
Comment (peggy-gov-bc-ca) [reply app/DynamicsAdapter/DynamicsAdapter.Web.Test/MockLoggerExtensions.cs:19]: the code following this comment is for .net core 2.1. Want to keep it there as a reminder.

File: app/DynamicsAdapter/DynamicsAdapter.Web/PersonSearch/PersonSearchController.cs [hunk 3]
```diff
@@ -226,6 +230,6 @@ public async Task<IActionResult> Rejected(Guid id, [FromBody]PersonSearchRejecte
             }
         }
 
- 
+
     }
 }
```
[the remark this answers is not in the document]
Comment (peggy-gov-bc-ca) [reply app/DynamicsAdapter/DynamicsAdapter.Web/PersonSearch/PersonSearchController.cs:235]: ok

File: app/DynamicsAdapter/DynamicsAdapter.Web.Test/MockLoggerExtensions.cs (hunk 1 above)
Comment (peggy-gov-bc-ca) [reply app/DynamicsAdapter/DynamicsAdapter.Web.Test/MockLoggerExtensions.cs:7]: This is not testing. This is an extension for verify log. 
It is used by SearchRequestJobTest and SearchResultServiceTest. When exception throws, it is used to verify the error is logged.

File: app/DynamicsAdapter/DynamicsAdapter.Web/PersonSearch/PersonSearchController.cs (hunk 3 above)
[the remark this answers is not in the document]
Comment (peggy-gov-bc-ca) [reply app/DynamicsAdapter/DynamicsAdapter.Web/PersonSearch/PersonSearchController.cs:235]: do not understand...

File: app/DynamicsAdapter/DynamicsAdapter.Web.Test/MockLoggerExtensions.cs (hunk 2 above)
[the remark this answers is not in the document]
Comment (rantifamilusi) [reply app/DynamicsAdapter/DynamicsAdapter.Web.Test/MockLoggerExtensions.cs:19]: We wouldn't be using .net core 2.1 and we can keep it in the read me if we want to do that.

[the remark this answers is not in the document]
Comment (peggy-gov-bc-ca) [reply app/DynamicsAdapter/DynamicsAdapter.Web.Test/MockLoggerExtensions.cs:19]: ok. will do that.

**Pull request merged**